### PR TITLE
Create debug.md

### LINKED ISSUE #254: Add debug guide - deployment failing with "failed to find provider binary azure"
**What steps did you take and what happened:**
1- Deploy basic AKS cluster in Azure using System Identity (not sure if System Identity is relevant)
2- Use helm to install the provider:
```
 helm install  csi-secrets-store secrets-store-csi-driver/secrets-store-csi-driver
```
3- Deploy provider:
```
apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
kind: SecretProviderClass
metadata:
  name: azure-kvname
spec:
  provider: azure                   
  parameters:
    usePodIdentity: "false"         # [OPTIONAL for Azure] if not provided, will default to "false"
    useVMManagedIdentity: "true"   # [OPTIONAL available for version > 0.0.4] if not provided, will default to "false"
id to specify which user assigned managed identity to use. If using a user assigned identity as the VM's managed identity, then specify the identity's client id. If empty, then defaults to use the system assigned identity on the VM
    keyvaultName: "my-keyvault"          # the name of the KeyVault
    objects:  |
      array:
        - |
          objectName: mysecret
          objectType: secret        # object types: secret, key or cert. For Key Vault certificates, refer to https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/docs/getting-certs-and-keys.md for the object type to use
    resourceGroup: "my-resources"            # [REQUIRED for version < 0.0.4] the resource group of the KeyVault
    subscriptionId: "$SUBID"         # [REQUIRED for version < 0.0.4] the subscription ID of the KeyVault
    tenantId: "$TENANTID"         
```
```
 kubectl apply -f .\provider.yaml 
````
4- Deploy the nginx sample config:
```
# This is a sample pod definition for using SecretProviderClass and service-principal for authentication with Key Vault
kind: Pod
apiVersion: v1
metadata:
  name: nginx-secrets-store-inline
spec:
  containers:
  - image: nginx
    name: nginx
    volumeMounts:
    - name: secrets-store-inline
      mountPath: "/mnt/secrets-store"
      readOnly: true
  volumes:
    - name: secrets-store-inline
      csi:
        driver: secrets-store.csi.k8s.io
        readOnly: true
        volumeAttributes:
          secretProviderClass: "azure-kvname"
```
``` 
kubectl apply -f .\nginx.yaml
 ````

5- Checking the status of the nginx deployment:
```
Events:
  Type     Reason       Age                            From                                        Message
  ----     ------       ----                           ----                                        -------
  Normal   Scheduled    <invalid>                      default-scheduler                           Successfully assigned default/nginx-secrets-store-inline to aks-agentpool-36317948-vmss000001
  Warning  FailedMount  <invalid> (x3 over <invalid>)  kubelet, aks-agentpool-36317948-vmss000001  MountVolume.SetUp failed for volume "secrets-store-inline" : rpc error: code = Unknown desc = failed to mount secrets store objects for pod default/nginx-secrets-store-inline, err: failed to find provider binary azure, err: stat /etc/kubernetes/secrets-store-csi-providers/azure/provider-azure: no such file or directory
```
**err: failed to find provider binary azure, err: stat /etc/kubernetes/secrets-store-csi-providers/azure/provider-azure: no such file or directory**

**What did you expect to happen:**
I expected the nginx container to come up normally.

**Anything else you would like to add:**
```
PS C:\Users\h\dev> kubectl get ds  -o jsonpath="{range .items[*]}{.spec.template.spec.containers[1].args}{'\n'}"
[--debug=true --endpoint=$(CSI_ENDPOINT) --nodeid=$(KUBE_NODE_NAME) --provider-volume=/etc/kubernetes/secrets-store-csi-providers --metrics-addr=:8080]
```

**Which access mode did you use to access the Azure Key Vault instance:**
[e.g. Service Principal, Pod Identity, User Assigned Managed Identity, System Assigned Managed Identity]

Tried with User Assigned Managed Identity and System Assigned Managed Identity

**Environment:**

- Secrets Store CSI Driver version: (use the image tag): latest
- Azure Key Vault provider version: (use the image tag): latest
- Kubernetes version: (use `kubectl version` and `kubectl get nodes -o wide`):
```
PS C:\Users\h\dev> kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.1", GitCommit:"d224476cd0730baca2b6e357d144171ed74192d6", GitTreeState:"clean", BuildDate:"2020-01-14T21:04:32Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.8", GitCommit:"73ec19bdfc6008cd3ce6de96c663f70a69e2b8fc", GitTreeState:"clean", BuildDate:"2020-09-17T04:17:08Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}

PS C:\Users\h\dev> kubectl get nodes -o wide
NAME                                STATUS   ROLES   AGE    VERSION   INTERNAL-IP   EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION      CONTAINER-RUNTIME
aks-agentpool-36317948-vmss000000   Ready    agent   130m   v1.18.8   10.240.0.4    <none>        Ubuntu 16.04.7 LTS   4.15.0-1096-azure   docker://19.3.12
aks-agentpool-36317948-vmss000001   Ready    agent   130m   v1.18.8   10.240.0.5    <none>        Ubuntu 16.04.7 LTS   4.15.0-1096-azure   docker://19.3.12
aks-agentpool-36317948-vmss000002   Ready    agent   130m   v1.18.8   10.240.0.6    <none>        Ubuntu 16.04.7 LTS   4.15.0-1096-azure   docker://19.3.12
```
- Cluster type: (e.g. AKS, aks-engine, etc): AKS

### PR DESCRIPTION
**Pull request by salehio**
Create debug.md to act as a quick index for fixing common issues with deployment based on symptoms.

**Reason for Change**:

Adding a debug.md to act as a knowledge base for errors and their associated solutions and debug steps.

**Requirements**

- [x] squashed commits
- [x] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

Fixes #254

**Please answer the following questions with yes/no**:

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
no

### CODE REVIEW

File: docs/debug.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Troubleshooting
+The following is a (growing) list of possible errors you might run in to when setting up the CSI Driver provider for auzre, along with recommendations for remediation.
```
Comment (aramase) [inline docs/debug.md:2]:
```suggestion
The following is a (growing) list of possible errors you might run in to when setting up the Azure Key Vault Provider for Secrets Store CSI Driver , along with recommendations for remediation.
```

Comment (aramase) [inline docs/debug.md:2]: nit: add new line

File: docs/debug.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+## Container deployment failing with "failed to find provider binary azure"
+### Symptom
+When attempting to install your service that leverages the newly created SecretProvider, your container deployment might not succeed. If you run `kubectl describe pod my_pod` you might see something like this:
```
Comment (aramase) [inline docs/debug.md:6]:
```suggestion
Application pod mount fails when deployed with `SecretProviderClass`. Running `kubectl describe pod <pod name>` contains the error `err: failed to find provider binary azure, err: stat /etc/kubernetes/secrets-store-csi-providers/azure/provider-azure: no such file or directory`
```

File: docs/debug.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+```
+Events:
+  Type     Reason       Age                            From                                        Message
+  ----     ------       ----                           ----                                        -------
+  Normal   Scheduled    <invalid>                      default-scheduler                           Successfully assigned default/nginx-secrets-store-inline to aks-agentpool-36317948-vmss000001
+  Warning  FailedMount  <invalid> (x3 over <invalid>)  kubelet, aks-agentpool-36317948-vmss000001  MountVolume.SetUp failed for volume "secrets-store-inline" : rpc error: code = Unknown desc = failed to mount secrets store objects for pod default/nginx-secrets-store-inline, err: failed to find provider binary azure, err: stat /etc/kubernetes/secrets-store-csi-providers/azure/provider-azure: no such file or directory
+```
+
+Particularly this part: **_err: failed to find provider binary azure_**
```
Comment (aramase) [inline docs/debug.md:15]: nit: we can remove this

File: docs/debug.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+### Explanation
```
Comment (aramase) [inline docs/debug.md:16]: nit: s/Explanation/Reason

File: docs/debug.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+This means the keyvault provider was not installed as part of the CSI driver installation.
```
Comment (aramase) [inline docs/debug.md:17]:
```suggestion
This means the Azure Key Vault provider was not installed as part of the installation.
```

File: docs/debug.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+### Solution
+If the CSI driver was installed via helm, uninstall it and reinstall from the correct package found [here](https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/charts/csi-secrets-store-provider-azure/README.md)
```
Comment (aramase) [inline docs/debug.md:19]: nit: s/CSI driver/Secrets Store CSI Driver

File: docs/debug.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+If installed via deployment files, follow the instructions [here](https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/docs/install-yamls.md) to install the keyvault provider
```
Comment (aramase) [inline docs/debug.md:21]: nit: s/keyvault provider/Azure Key Vault Provider for Secrets Store CSI Driver

File: docs/debug.md (hunk 1 above)
Comment (salehio) [reply docs/debug.md:2]: Below the Troubleshooting header or do you mean another below the intro?

**Comment by aramase** [comment]
@salehio Thank you for the PR. We're in the process of moving our docs to a static docs site. I'll add a troubleshooting guide to the website once it's up. Would it be ok to close this PR for now?

**Comment by salehio** [comment]
@aramase Yeah, feel free to do with it as you please.

**Comment by aramase** [comment]
Thank you @salehio

**Pull request closed without merging**